app/src/main/java/com/andrew/tasky/agenda/presentation/notifications/AgendaNotificationService.kt

```diff
@@ -0,0 +1,49 @@
+package com.andrew.tasky.agenda.presentation.notifications
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.NotificationCompat
+import androidx.navigation.NavDeepLinkBuilder
+import com.andrew.tasky.R
+import com.andrew.tasky.core.MainActivity
+
+class AgendaNotificationService(
+    private val context: Context
+) {
+
+    private val notificationManager = context
+        .getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+    fun showEventNotification(title: String, description: String) {
+        val eventNavIntent = NavDeepLinkBuilder(context)
+            .setGraph(R.navigation.my_nav)
+            .setDestination(R.id.reminderDetailFragment)
+            .createPendingIntent()
+
+        val intent = Intent(context, MainActivity::class.java)
+        intent.putExtra(EVENT_NAV_INTENT, eventNavIntent)
```

Comment:
The key here can be the same for all types of agenda items, so doesn't need to be specific to events